### PR TITLE
[WIP] use rsync instead of copymitter

### PR DESCRIPTION
**Pull request by ericfranz**

### CODE REVIEW

**Comment by ericfranz** [comment]
This failed to solve the problem.

**Comment by ericfranz** [comment]
https://discourse.osc.edu/t/directory-copy-and-paste-issue-with-lustre/171

**Comment by ericfranz** [comment]
Using this branch actually worked. We might want to clean up the comments and add progress display if the rsync module supports that like the copymitter did.

File: lib/cloudcmd/lib/server/rest.js [hunk 1]
```diff
@@ -24,7 +24,8 @@
         copymitter  = require('copymitter'),
```
Comment (MorganRodgers) [inline lib/cloudcmd/lib/server/rest.js]: If `copymitter` is no longer used then remove the `require`.

File: lib/cloudcmd/lib/server/rest.js [hunk 2]
```diff
@@ -338,25 +339,33 @@
...
         });
-        
-        cp.on('end', function() {
-            process.stdout.write('\n');
-            fn(error);
+
+        rsync.execute(function(error, stdout, stderr) {
+          fn(error);
         });
+
```
Comment (MorganRodgers) [inline lib/cloudcmd/lib/server/rest.js]: TODO: remove the commented out code.

File: lib/cloudcmd/lib/server/rest.js [hunk 3]
```diff
@@ -338,25 +339,33 @@
...
         var error,
             tmpl    = '\r copy {{ from }} {{ to }} {{ count }}%',
-            cp      = copymitter(from, to, names);
-         
-        cp.on('error', function(e) {
-            error = e;
-            cp.abort();
-        });
-        
-        cp.on('progress', function(count) {
```
Comment (MorganRodgers) [inline lib/cloudcmd/lib/server/rest.js]: Ideally the progress callback will be re-implemented. The command `rsync -a --info=progress2 --progress $SRC $DST` produces output like:

```sh
neo-dashboard/node_modules/path-parse/.travis.yml
     18,197,451  80%  424.41kB/s    0:00:41 (xfr#3925, ir-chk=1005/6295)
```

Giving us a source file, and a percent complete. A caveat is that `progress2` was introduced "recently": https://serverfault.com/a/441724. I am not sure what Centos 6 has available, but MacOS for example uses `rsync` 2.6.9; it is available in Centos 7.

Comment (MorganRodgers) [reply lib/cloudcmd/lib/server/rest.js]: If we proceed with this solution we'll want to update our list of external programs to ensure that `rsync` is both installed and has a compatible version.

**Comment by MorganRodgers** [comment]
The best solution needs to preserve file modes / permissions etc.

**Comment by ericfranz** [comment]
Good idea to compare stat attributes of file prior to and after a copy using cp

**Comment by MorganRodgers** [comment]
I just tried upgrading `copymitter` and a simple `yarn upgrade` does not work.

File: lib/cloudcmd/lib/server/rest.js (hunk 3 above)
Comment (ericfranz) [reply lib/cloudcmd/lib/server/rest.js]: FWIW the rsync node module does appear to provide progress events though perahps it is based on later versions of rsync like you have identified above?

**Comment by ericfranz** [comment]
>I just tried upgrading copymitter and a simple yarn upgrade does not work.

How did you confirm? Do you have access to a Lustre file system? Or you ran into other problems?

**Comment by MorganRodgers** [comment]
Sorry, but "did not work" I meant to say that the dependency cannot be upgraded.

**Comment by MorganRodgers** [comment]
Given that this PR does not implement progress messages and #202 does, I think this should be closed. @ericfranz and @johrstrom thoughts?

**Comment by ericfranz** [comment]
closing as duplicate of #202

**Comment by ericfranz** [comment]
Closing as duplicate of https://github.com/OSC/ood-fileexplorer/pull/208

**Pull request closed without merging**